doc/2/guides/main-concepts/1-api/index.md

```diff
@@ -8,3 +8,461 @@ order: 100
+
+List of [Standard Kuzzle Error](/core/2/api/some-link)
+
+**Example:** _Receving a `network.http.url_not_found`_
+
+```bash
+curl "localhost:7512/_i_am_not_a_valid_url?pretty"
+
+# {
+#   "requestId": "cbafaf6e-0464-4787-a2b4-633739e7c677",
+#   "status": 404,
+#   "error": {
+#     "message": "API URL not found: /_i_am_not_a_valid_url.",
+#     "status": 404,
+#     "id": "network.http.url_not_found",
+#     "code": 50397191
+#   },
+#   "controller": null,
+#   "action": null,
+#   "collection": null,
+#   "index": null,
+#   "volatile": null,
+#   "result": null
+# }
+```
+
+### id
+
+The `id` property is unique to each type of error that can be returned, and is built by concatenating the following information:
+
+* Domain: from where the error comes from (API, network, plugin, ...)
+* Subdomain: what kind of error it is (assertion, runtime, ...)
+* Error: the error itself
+
+For instance:
+* `api.assert.missing_argument` is an assertion error triggered by the API because of a missing argument
+* `network.http.url_not_found` is a HTTP error triggered by the network layer, because a requested URL couldn't be found
+
+The complete list of API errors is available [here](/core/2/api/essentials/error-codes/).
+
+### code
+
+The `code` property is a 32-bits integer representation of the unique `id` error identifier, detailed above.
+
+It's meant to be **used by low-level languages** to efficiently catch specific error codes and act on them.
+
+Code format:
+- Domain: ranges from `00` to `FF` (1 byte)
+- Subdomain: ranges from `00` to `FF` (1 byte)
+- Error: ranges from `0000` to `FFFF` (2 bytes)
+
+The complete list of API errors is available [here](/core/2/api/essentials/error-codes/).
+
+## Volatile Data
+
+All requests accept a `volatile` object in parameter.
+
+The content of this object is not meant to be used directly: it has no impact on the request itself.
+
+Still, volatile data are not completely ignored by Kuzzle, and they have a few uses.
+
+### Request Context
+
+Volatile data can be used to **provide additional context about a request**; this allows **extended logs**, **application metadata**, and so on. Many use cases benefit from being able to pass context data, without any direct impact to requests themselves.
+
+Lastly, if a request triggers a [document notification](/core/2/guides/main-concepts/1-api#document-notification), then its **volatile data are included in the notification content**. This allows real-time subscribers to get elements of context about changes made to documents, if needs be.
+
+::: info
+By default, SDKs includes two fields in the request volatile data:
+ - `sdkInstanceId`: unique identifier for this SDK instance
+ - `sdkName`: SDK name and version
+:::
+
+### Realtime Subscription Context
+
+There is one special case, where volatile data are stored by Kuzzle for a later use, instead of being completely ignored: whenever a client make a new real-time subscription.
```

Comment:
```suggestion
There is one special case, where volatile data are stored by Kuzzle for a later use, instead of being completely ignored: whenever a client makes a new real-time subscription.
```